### PR TITLE
MAINT-36799 : Cannot edit document with onlyoffice when the space name contains special chars

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**
**Problem :**  Cannot edit document with onlyoffice when the space name contains special chars .
**Solution :**  clean the name of drive.

### CODE REVIEW

File: webapp/src/main/webapp/js/onlyoffice.js [hunk 1]
```diff
@@ -1086,6 +1086,10 @@
       var drive = config.editorPage.displayPath.split(':')[0].replace(/\  /g, ' ');
       var $bar = $("#editor-drawer");
       if (drive.startsWith('spaces/')) {
+        // Clean name of drive
+        const urlParams = new URLSearchParams(config.explorerUrl);
```
Comment (ahamdi) [inline webapp/src/main/webapp/js/onlyoffice.js:1090]: Could you check and fix the drive name directly from the portlet instead of JS ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: webapp/src/main/webapp/js/onlyoffice.js (hunk 1 above)
Comment (Rdidmohamedelhadi) [reply webapp/src/main/webapp/js/onlyoffice.js:1090]: ok I will do that

**Review by ahamdi — CHANGES_REQUESTED** [review]
Add a UT to make sure we get the drive of the file correctly

**Review by ahamdi — APPROVED** [review]

File: webapp/src/main/webapp/js/onlyoffice.js [hunk 2]
```diff
@@ -1083,7 +1083,7 @@
     this.initBar = function(config) {
       config.editorPage.displayPath = decodeURI(config.editorPage.displayPath);
       config.path = decodeURI(config.path);
-      var drive = config.editorPage.displayPath.split(':')[0].replace(/\  /g, ' ');
+      var drive = config.editorPage.drive;
```
Comment (ahamdi) [inline webapp/src/main/webapp/js/onlyoffice.js:1086]: Could you manage the case if drive is null? Dince the function getdrive may return null

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: webapp/src/main/webapp/js/onlyoffice.js (hunk 2 above)
Comment (Rdidmohamedelhadi) [reply webapp/src/main/webapp/js/onlyoffice.js:1086]: ok

**Pull request merged**